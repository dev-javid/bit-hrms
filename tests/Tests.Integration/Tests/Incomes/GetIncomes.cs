namespace Tests.Integration.Tests.Incomes;

public class GetIncomes(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
{
    private const string Route = "/api/incomes?page=1&limit=10";

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public async Task Should_return_data()
    {
        await LoginAsCompanyAdminAsync();
        await FeedDataAsync("Tests/Incomes/GetIncomes.sql");

        HttpResponseMessage response = await Client.GetAsync(Route);
        response.StatusCode.Should().Be(HttpStatusCode.OK);

        var result = await response.Content.ReadAsStringAsync();
        Approvals.VerifyJson(result);
    }
}
