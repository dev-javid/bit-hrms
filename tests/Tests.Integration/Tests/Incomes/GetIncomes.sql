INSERT INTO income_sources (id, company_id, name, description, created_on, created_by, modified_by, modified_on)
VALUES (1,  999, 'Income Source 1', 'Description 1',  '2024-05-03 03:51:44.770234-07', 100, null, null),
       (2,  999, 'Income Source 2', 'Description 2',  '2024-06-03 03:51:44.770234-07', 100, null, null);

INSERT INTO incomes (id, company_id, income_source_id, amount, documents, remarks, created_on, created_by, modified_by, modified_on)
VALUES (1, 999, 1, decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-03 03:51:44.770234-07', 100, null, null),
       (2, 999, 2, decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), ARRAY['test.png', 'test2.png'], null, '2024-05-04 03:51:44.770234-07', 100, null, null);

