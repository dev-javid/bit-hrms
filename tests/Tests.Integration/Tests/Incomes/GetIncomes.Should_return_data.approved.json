{
    "page": 1,
    "limit": 10,
    "total": 2,
    "items": [
        {
            "incomeId": 2,
            "amount": 1200,
            "incomeSourceName": "Income Source 2",
            "documents": [
                "http://localhost/media/Incomes/2/test.png",
                "http://localhost/media/Incomes/2/test2.png"
            ],
            "remarks": null,
            "createdOn": "2024-05-04T10:51:44.770234Z"
        },
        {
            "incomeId": 1,
            "amount": 1000,
            "incomeSourceName": "Income Source 1",
            "documents": [
                "http://localhost/media/Incomes/1/test.png",
                "http://localhost/media/Incomes/1/test2.png"
            ],
            "remarks": null,
            "createdOn": "2024-05-03T10:51:44.770234Z"
        }
    ]
}